### ISSUE TITLE
[Bug]: Radio control accessibility

### ISSUE THREAD

**Issue by APJohns** [storybookjs/storybook#24036]
### Describe the bug

The radio control uses different name values for each input. These names should be the same to link the inputs together in one radio group. Otherwise screen readers will see each radio input as its own group of 1 input. The keyboard interactions are also effected. You should tab to radio group and use arrows to select an option/input. Currently you tab through each radio input and press space to select. This goes against the expected behavior of a radio group.

### To Reproduce

1. Go to a story of a component with a radio control.
2. Tab to radio control.
3. Notice you need to press tab to focus the input and space to select instead of selecting using arrow keys.
    - Also notice screen readers read each input as a radio group containing 1 input.

**Comment by APJohns** [comment]
Looks like we need to use `controlId` instead of `id` for the name value in the code linked below.
https://github.com/storybookjs/storybook/blob/115d439da3f8e418e306d10fc2026383a130e4be/code/ui/blocks/src/controls/options/Radio.tsx#L55-L66

I'd put up a PR but I'm running into some errors when setting up a local environment and wouldn't be able to test my changes.

**Comment by TheSeydiCharyyev** [comment]
This appears to be resolved in the current `next` branch. `RadioControl` now sets `name={controlId}` on each `<input type="radio">`, so all options in one group share the same `name`:

https://github.com/storybookjs/storybook/blob/next/code/addons/docs/src/blocks/controls/options/Radio.tsx#L82-L90

The options are also wrapped in a `<fieldset>` with an sr-only `<legend>`, which gives the group an accessible name and makes the browser apply native radio-group keyboard semantics (Tab to enter the group, arrow keys to move between options). Screen readers announce a single group with N inputs in my testing on Windows 11 / Chrome.
